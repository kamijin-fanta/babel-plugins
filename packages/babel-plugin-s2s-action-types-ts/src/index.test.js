import plugin from './index';
import pluginTester from 'babel-plugin-tester';
import path from 'path';

pluginTester({
  plugin,
  fixtures: path.resolve(__dirname, '__fixtures__'),
  pluginOptions: {
    removePrefix: 'packages/babel-plugin-s2s-action-types-ts/src/__fixtures__/',
  },
});

pluginTester({
  plugin,
  pluginOptions: {
    usePrefix: false,
  },
  snapshot: true,
  tests: [
    {
      title: 'usePrefix: false',
      code: `export type Action = Foo`,
    },
  ],
});
