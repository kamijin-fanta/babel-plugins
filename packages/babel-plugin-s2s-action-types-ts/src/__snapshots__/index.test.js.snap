// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty Actions 1`] = `
"
/* empty */

      ↓ ↓ ↓ ↓ ↓ ↓

/* empty */
export type Action = ;

/***** Do not edit below this line *****/
export const enum Actions {}
"
`;

exports[`usePrefix: false 1`] = `
"
export type Action = Foo

      ↓ ↓ ↓ ↓ ↓ ↓

export type Action = Foo;

export interface Foo {
  type: Actions.Foo;
}

/***** Do not edit below this line *****/
export const enum Actions {
  Foo = \\"Foo\\",
}
"
`;
